test/new-e2e/tests/agent-metric-logs/log-agent/linux-log/file-tailing/auto_multi_line_test.go

```diff
@@ -47,51 +50,90 @@ func TestAutoMultiLineSuite(t *testing.T) {
 	// Alternate between sending single line logs and multi line logs
-	for i := 0; i < 700; i++ {
-		timestamp := time.Now().Format(time.RFC3339)
+	for i := 0; i < count; i++ {
 		if i%2 == 0 {
-			message = fmt.Sprintf("%s | %s", timestamp, singleLineLog)
+			message = fmt.Sprintf("%s | %s", prefix, singleLineLog)
 		} else {
-			message = fmt.Sprintf("%s | %s", timestamp, multiLineLog)
+			message = fmt.Sprintf("%s | %s", prefix, multiLineLog)
 		}
-		cmd := fmt.Sprintf("echo '%s' | sudo tee -a %s", message, logFilePath)
-		s.Env().RemoteHost.MustExecute(cmd)
+		s.Env().RemoteHost.AppendFile("linux", logFilePath, []byte(message))
 	}
 }
 
-func (s *AutoMultiLineSuite) BeforeTest(suiteName, testName string) {
-	s.BaseSuite.BeforeTest(suiteName, testName)
+func (s *MultiLineSuite) detectsAutoMultiLine() {
+	agentOptions := []agentparams.Option{
+		agentparams.WithLogs(),
+		agentparams.WithIntegration("custom_logs.d", agentAutoMultiLineConfig),
+	}
+	// Update env to enable auto_multi_line_detection
+	s.UpdateEnv(awshost.Provisioner(awshost.WithAgentOptions(agentOptions...)))
 
-	// Create a new log folder location
-	s.Env().RemoteHost.MustExecute(fmt.Sprintf("sudo mkdir -p %s", utils.LinuxLogsFolderPath))
+	// auto_multi_line_detection uses the first 500 logs to detect a pattern
+	s.generateMultilineLogs(time.Now().Format(time.RFC3339), 700)
 
-	s.Env().RemoteHost.Execute(fmt.Sprintf("sudo touch %s", logFilePath))
-	s.Env().RemoteHost.Execute(fmt.Sprintf("sudo chmod +r %s", logFilePath))
+	client := s.Env().FakeIntake.Client()
 
-	s.generateMultilineLogs()
-}
+	// Raw string since '\n' literal will be in the log.Message
+	content := `This is a multi\nline log`
+	service := "hello"
 
-func (s *AutoMultiLineSuite) AfterTest(suiteName, testName string) {
-	s.BaseSuite.AfterTest(suiteName, testName)
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		names, err := client.GetLogServiceNames()
+		if !assert.NoErrorf(c, err, "Error found: %s", err) {
+			return
+		}
 
-	s.Env().RemoteHost.Execute(fmt.Sprintf("sudo rm -rf %s", utils.LinuxLogsFolderPath))
+		// Auto Multiline is working if the log message contains the complete log contents with newlines
+		logs, err := client.FilterLogs(service, fi.WithMessageContaining(content))
+		if !assert.NoErrorf(c, err, "Error found: %s", err) {
+			return
+		}
+
+		assert.NotEmpty(c, logs, "Expected at least 1 log with content: '%s', from service: %s but received %s logs.", content, names, logs)
+	}, 2*time.Minute, 10*time.Second)
 }
 
-func (s *AutoMultiLineSuite) ContainsLogWithNewLines() {
+func (s *MultiLineSuite) detectsPatternMultiLine() {
+	agentOptions := []agentparams.Option{
+		agentparams.WithLogs(),
+		agentparams.WithIntegration("custom_logs.d", multiLineLogPatternConfig),
+	}
+	// Update env to add log_processing_rules with multi_line pattern
+	s.UpdateEnv(awshost.Provisioner(awshost.WithAgentOptions(agentOptions...)))
+
+	s.generateMultilineLogs("custom-log-prefix", 100)
```

Comment:
Shouldn't this be `fake-log-prefix`? as specified in the pattern by the yaml config?

Reply:
wait yes that is strange lol. wondering how the test passed, going to take another look

Reply:
The problem was that `multi_line` aggregates all lines into a single entry until the pattern is found / line limit is reached but in the previous code, the pattern was never found due to the mismatched prefix in the yaml, so multiple logs were combined which made `fakeintake.WithMessageContaining(content)` incorrectly pass.

I updated the code to use a regex pattern instead to ensure that the log consists entirely of only one multi line log